moduleframework/tools/dockerlint.py

```diff
@@ -94,7 +94,10 @@ def test_chained_run_rest_commands(self):
         self.assertTrue(self.dp.check_chained_run_rest_commands())
 
     def test_helpmd_is_present(self):
-        self.assertTrue(self.dp.check_helpmd_is_present())
+        helpmd_present = self.dp.check_helpmd_is_present()
+        if not helpmd_present:
+            self.log.warn("help.md is not present in Dockerfile")
+        self.assertTrue(True)
```

Comment:
I don't know avocado, but is this really needed? Does every test need to be asserted?

Reply:
I have switched the test from FAIL and PASS to PASS and WARN, because of help.md is not important.
Otherwise I will revert it.

Reply:
This `self.assertTrue(True)` is very bad from my PoV,  maybe easy to understand could be with comment:
```
if not helpmd_present:
    # you have to have at lease one assert to report state as __WARN__
    self.assertTrue(True) 
    self.log.warn("help.md is not present in Dockerfile")
else:
    self.assertTrue(helpmd_present)
```
or maybe even better create and add some decorator around self.assert*

like
```
class X...
    def mark_as_warn(self,function,*args, **kwargs):
       try:
          function(*args, **kwargs)
       except Exception as e:
          self.assertTrue(True)
          self.log.warn("Warning raised: %s" % e)

and then use inside test like: ` self.mark_as_warn(self.assertTrue, False)`

what do you think? We can add this to AvocadoTest class https://github.com/fedora-modularity/meta-test-family/blob/devel/moduleframework/avocado_testers/avocado_test.py#L40

and then you can use this function everywhere you will use assert but expect to return just warn

Reply:
seems like a bug in avocado (that you need to assert in order to make a warn)

Reply:
I've tried that and it is not important to have there this assertTrue(True) anyhow.
@phracek you can have test without any assert and it is possible to print warn well, it also means that fixes like https://github.com/fedora-modularity/meta-test-family/pull/150/files#diff-89b2554ab29801f3d6668f1e0fa567f9R62 are important, because now, these tests pass, although they do nothing, no asserts means pass by default, what is expected behavior.

```
from avocado import Test

class X(Test):

    def testNo(self):
        pass

    def testTrue(self):
        self.assertTrue(True)

    def testFalse(self):
        self.assertTrue(False)

    def testWarn(self):
        self.log.warn("pass but bad")
```
returns:
```
 (1/4) test.py:X.testNo: PASS (0.01 s)
 (2/4) test.py:X.testTrue: PASS (0.02 s)
 (3/4) test.py:X.testFalse: FAIL (0.04 s)
 (4/4) test.py:X.testWarn: WARN (0.00 s)
```
so your code could be writen like: 
```
if not helpmd_present:
    self.log.warn("help.md is not present in Dockerfile")
else:
    self.assertTrue(helpmd_present)
```

or write some generic method and use it to transform assert to warn and use it then as described:
```
def mark_as_warn(self,function,*args, **kwargs):
       try:
          function(*args, **kwargs)
       except AssertionError as e:
          self.log.warn("Warning raised: %s" % e)
```